common/src/main/java/tachyon/client/WorkerBlockMasterClient.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.MasterClientBase;
+import tachyon.conf.TachyonConf;
+import tachyon.thrift.BlockMasterService;
+import tachyon.thrift.Command;
+import tachyon.thrift.NetAddress;
+
+
+
+/**
+ * A wrapper for the thrift client to interact with the block master, used by tachyon worker.
+ * <p/>
+ * Since thrift clients are not thread safe, this class is a wrapper to provide thread safety, and
+ * to provide retries.
+ */
+public final class WorkerBlockMasterClient extends MasterClientBase {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private BlockMasterService.Client mClient = null;
+
+  /**
+   * Creates a new block master client for the worker.
+   *
+   * @param masterAddress the master address
+   * @param executorService the executor service
+   * @param tachyonConf the Tachyon configuration
+   */
+  public WorkerBlockMasterClient(InetSocketAddress masterAddress, ExecutorService executorService,
+      TachyonConf tachyonConf) {
+    super(masterAddress, executorService, tachyonConf);
+  }
+
+  @Override
+  protected String getServiceName() {
+    return Constants.BLOCK_MASTER_SERVICE_NAME;
+  }
+
+  @Override
+  protected void afterConnect() {
+    mClient = new BlockMasterService.Client(mProtocol);
+  }
+
+  /**
+   * Commits a block on a worker.
+   *
+   * @param workerId the worker id committing the block
+   * @param usedBytesOnTier the amount of used bytes on the tier the block is committing to
+   * @param tier the tier the block is being committed to
+   * @param blockId the block id being committed
+   * @param length the length of the block being committed
+   * @throws IOException if an I/O error occurs
+   */
+  public synchronized void workerCommitBlock(long workerId, long usedBytesOnTier, int tier,
```

Comment:
Could you remove the `worker` prefix on the methods (but not at the thrift layer since it still makes sense there)?